inspirehep/modules/workflows/utils.py

```diff
@@ -214,3 +216,28 @@ def retrieve_root_json(record_uuid, source):
         WorkflowsRecordSources.source == source
     ).one_or_none()
     return entry.json if entry else {}
+
+
+def store_head_json(record_uuid, json):
+    """Store the head json for a given record.
+
+    Given the actual record in ``json`` and is key ``record_uuid``
+    store this information so that it can be retrieved later.
+    """
+    record = InspireRecord.get_record(record_uuid)
+    record.update(json)
+    # Commit any changes to record
+    record.commit()
```

Comment:
This is not really committing into the DB, so we need to ensure that the actual commit happens here or at least within the larger `merge_article` workflow task.